### ISSUE TITLE
SQL Server scaffolding generates invalid anonymous types for indexes (rc1-final)

### ISSUE THREAD

**Issue by AlekseyMartynov** [aspnet/EntityFramework#3847]
I use the NORTHWND database restored from https://northwinddatabase.codeplex.com, and use the following command to scaffold entity classes:

    dnx EntityFramework.Commands dbcontext scaffold "data source=.\SQLEXPRESS;initial catalog=NORTHWND;integrated security=True" EntityFramework.MicrosoftSqlServer -a -c NorthwindContext -o Models\Northwind

Design package is EntityFramework.MicrosoftSqlServer.Design 7.0.0-rc1-final

Generated classes fail to compile with "An anonymous type cannot have multiple properties with the same name".

Problematic source lines:
```c#
entity.HasIndex(e => new { e.CompanyName, e.CompanyName }).HasName("CompanyName");
entity.HasIndex(e => new { e.PostalCode, e.PostalCode, e.PostalCode }).HasName("PostalCode");
```

**Comment by lajones** [comment]
Hi Aleksey. Thanks for reporting this. It looks like there's a problem in the code which uniquifies property names and/or creates indexes. We'll look into it.

**Comment by lajones** [comment]
This was a consequence of us conflating indexes which had the same name but were on different tables. This was fixed by #3934.

**Comment by dumbledad** [comment]
Related stackoverflow question: [Entity Framework 7: An anonymous type cannot have multiple properties with the same name](http://stackoverflow.com/a/34221476/575530)
